src/puppetlabs/trapperkeeper/services/scheduler/job.clj

```diff
@@ -15,26 +15,55 @@
   []
   [[] (atom {})])
 
+(defn- recurring?
+  [options]
+  (or (contains? options :interval) (contains? options :interspaced)))
+
+(defn- calculate-next-execution-time
+  [context options]
+  (if (contains? options :interspaced)
+    (Date. ^Long (+ (System/currentTimeMillis) (:interspaced options)))
+    (.getFireTimeAfter (.getTrigger context) (Date.))))
+
+(defn- should-skip?
+  [context options]
+  (when (contains? options :interval)
+    (let [interval-ms (:interval options)
+          now (Date.)
+          now-ms (.getTime now)
+          next-fire-time (.getNextFireTime (.getTrigger context))
+          next-fire-time-ms (.getTime next-fire-time)]
+      (cond
```

Comment:
There is a "getScheduledFireTime" in the execution context that might be a better way of determining this.  Will investigate briefly.